jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-server/src/main/java/org/eclipse/jetty/ee10/websocket/jakarta/server/config/JakartaWebSocketConfiguration.java

```diff
@@ -35,6 +35,6 @@ public JakartaWebSocketConfiguration()
         protectAndExpose("org.eclipse.jetty.ee10.websocket.servlet."); // For WebSocketUpgradeFilter
         protectAndExpose("org.eclipse.jetty.ee10.websocket.jakarta.server.config.");
         protectAndExpose("org.eclipse.jetty.ee10.websocket.jakarta.client.JakartaWebSocketClientContainerProvider");
-        protectAndExpose("org.eclipse.jetty.ee10.websocket.jakarta.client.JakartaWebSocketShutdownContainer");
+        protectAndExpose("org.eclipse.jetty.ee10.websocket.jakarta.client.internal.JakartaWebSocketShutdownContainer");
```

Comment:
Feels wrong that we protect and expose an internal class, and not sure it really works in JPMS.
I think it's best to make it non-internal like the other one.

Reply:
Yeah the problem is if its non-internal we have a transitive dependency for the servlet api for the client module.
As the shutdown container extends `ServletContainerInitializer` and `ServletContextListener`.

Reply:
It is a strange case that it needs to be exposed so it can be discovered by the service loader in the webapps classloader when running SCIs.
I wonder if now we have an environment classloader, if perhaps we can look for SCIs at that level that are not themselves exposed to the webapp?

Reply:
Ugh, why do we need a `ServletContainerInitializer` in the client?!?

Reply:
So if a client is started by the application inside a web app, we can clean it up when the context stops, else it leaks and holds the context in memory.

Reply:
I thought we have a special jar classifier for that, like we have for the HTTP client?

Reply:
Okay, after discussion this can remain as it, provided we look at it from the point of view of the environment classloader, rather than the webapp classloader.